homeassistant/components/linkplay/media_player.py

```diff
@@ -0,0 +1,240 @@
+"""Support for LinkPlay media players."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from linkplay.bridge import LinkPlayBridge, LinkPlayMultiroom
+from linkplay.consts import LoopMode, PlayingStatus
+from linkplay.controller import LinkPlayController
+from linkplay.exceptions import LinkPlayException
+
+from homeassistant.components import media_source
+from homeassistant.components.media_player import (
+    BrowseMedia,
+    MediaPlayerDeviceClass,
+    MediaPlayerEntity,
+    MediaType,
+    RepeatMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from .const import (
+    BRIDGE_DISCOVERED,
+    CONTROLLER,
+    DEFAULT_FEATURES,
+    DOMAIN,
+    EQUALIZER_MAP,
+    EQUALIZER_MAP_INV,
+    REPEAT_MAP,
+    REPEAT_MAP_INV,
+    SEEKABLE_FEATURES,
+    SOURCE_MAP,
+    SOURCE_MAP_INV,
+    STATE_MAP,
+)
+from .utils import get_active_multiroom, get_info_from_project
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up a media player from a config entry."""
+
+    @callback
+    def add_entity(bridge: LinkPlayBridge) -> None:
+        async_add_entities([LinkPlayMediaPlayerEntity(bridge)])
+
+    entry.async_on_unload(async_dispatcher_connect(hass, BRIDGE_DISCOVERED, add_entity))
+
+
+class LinkPlayMediaPlayerEntity(MediaPlayerEntity):
+    """Representation of a LinkPlay media player."""
+
+    _bridge: LinkPlayBridge
+
+    def __init__(self, bridge: LinkPlayBridge) -> None:
+        """Initialize the LinkPlay media player."""
+        self._bridge = bridge
+
+        self._attr_unique_id = self.entity_id = f"{DOMAIN}.{self._bridge.device.uuid}"
+        self._attr_name = self._bridge.device.name
```

Comment:
Set media player to track device name:
```suggestion
        self._attr_has_entity_name = True
        self._attr_name = None
```